exercises/simple-cipher/.meta/src/reference/java/Cipher.java

```diff
@@ -32,23 +32,29 @@ public String getKey() {
     }
 
     public String encode(String plainText) {
-        String copy = plainText;
         StringBuilder ciphertext = new StringBuilder(plainText.length());
+        String sameLengthKey = getKeySameLengthAsPlaintext(plainText, key);
 
-        while (copy.length() > 0) {
-            for (int i = 0; i < Math.min(copy.length(), key.length()); i++) {
-                ciphertext.append(encodeCharacter(copy, i));
-            }
-            copy = "";
-            for (int i = ciphertext.length(); i < plainText.length(); i++) {
-                copy = copy + plainText.charAt(i);
-            }
+        for (int index = 0; index < plainText.length(); index++) {
+            ciphertext.append(encodeCharacter(plainText, index, sameLengthKey));
         }
 
         return ciphertext.toString();
     }
 
-    private char encodeCharacter(String plainText, int index) {
+    private String getKeySameLengthAsPlaintext(String plainText, String key) {
```

Comment:
I think this method actually returns a key that is at least as long as the plaintext? Which, when I think about it, should be fine and is probably better than trying to get it to be exactly the same length 🙂 But it's probably good to reflect that in the method name? So something like `getSufficientlyLongKeyForPlaintext` or something? 🙂

Reply:
Good point